istioctl/cmd/workload.go

```diff
@@ -0,0 +1,399 @@
+	temp, err := ioutil.TempDir("./", outputName)
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(temp)
+
+	if err = createClusterEnv(wg, temp); err != nil {
+		return err
+	}
+	if err = createCertsTokens(kubeClient, wg, temp); err != nil {
+		return err
+	}
+	if err = createMeshConfig(kubeClient, wg, clusterID, revision, temp); err != nil {
+		return err
+	}
+	if err = Tar(temp, outputName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write cluster.env into the given directory
+func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	we := wg.Spec.Template.Spec
+	for _, v := range we.Ports {
+		ports = append(ports, fmt.Sprint(v))
+	}
+	// default attributes and service name, namespace, ports, service account, service CIDR
+	clusterEnv := map[string]string{
+		"ISTIO_CP_AUTH":       "MUTUAL_TLS",
+		"ISTIO_INBOUND_PORTS": strings.Join(ports, ","),
+		"ISTIO_NAMESPACE":     wg.Namespace,
+		"ISTIO_PILOT_PORT":    "15012",
+		"ISTIO_SERVICE":       fmt.Sprintf("%s.%s", wg.Name, wg.Namespace),
+		"ISTIO_SERVICE_CIDR":  "*",
+		"SERVICE_ACCOUNT":     we.ServiceAccount,
+	}
+	return ioutil.WriteFile(dir+"/cluster.env", []byte(mapToString(clusterEnv)), tempPerms)
+}
+
+// Get and store the needed certificate and token. The certificate comes from the `istio-ca-root-cert`, and
+// the token is generated by kubectl under the workload group's namespace and service account
+// OSS requires both while ASM (will eventually) require only the token
+func createCertsTokens(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, dir string) error {
+	rootCert, err := kubeClient.CoreV1().ConfigMaps(wg.Namespace).Get(context.Background(), "istio-ca-root-cert", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if err = ioutil.WriteFile(dir+"/root-cert", []byte(rootCert.Data["root-cert.pem"]), tempPerms); err != nil {
+		return err
+	}
+
+	token := &authenticationv1.TokenRequest{
+		Spec: authenticationv1.TokenRequestSpec{
+			Audiences:         []string{"istio-ca"},
+			ExpirationSeconds: &tokenDuration,
+		},
+	}
+	namespace, serviceAccount := wg.Namespace, wg.Spec.Template.Spec.ServiceAccount
+	tokenReq, err := kubeClient.CoreV1().ServiceAccounts(namespace).CreateToken(context.Background(), serviceAccount, token, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(dir+"/istio-token", []byte(tokenReq.Status.Token), tempPerms); err != nil {
+		return err
+	}
+	return nil
+}
+
+func createMeshConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, dir string) error {
+	istioCM := "istio"
+	// Case with multiple control planes
+	if revision != "" {
+		istioCM = fmt.Sprintf("%s-%s", istioCM, revision)
+	}
+	istio, err := kubeClient.CoreV1().ConfigMaps("istio-system").Get(context.Background(), istioCM, metav1.GetOptions{})
```

Comment:
All istioctl commands take a `--istioNamespace` flag, should use that instead of hardcoded istio-system